ssl/statem/statem_srvr.c

```diff
@@ -3627,7 +3627,7 @@ static int create_ticket_prequel(SSL *s, WPACKET *pkt, uint32_t age_add,
      */
     if (!WPACKET_put_bytes_u32(pkt,
                                (s->hit && !SSL_IS_TLS13(s))
-                               ? 0 : s->session->timeout)) {
+                               ? 0 : (uint32_t)s->session->timeout)) {
```

Comment:
`session->timeout` was a `long` value type, but it is still used as an `uint32_t`, when put into protocol packet.
So a promotion to `uint32_t` instead of `time_t` could be smarter ?

Reply:
There may be some asn1 session serialisation issue to handle.

Reply:
IMHO, Best to keep internal use at the proper type (@paulidale said to convert these to time_t, as that's what it represents), when convert as necessary to necessary type (in this case a 32-bit type) for external use

Reply:
> `session->timeout` was a `long` value type, but it is still used as an `uint32_t`, when put into protocol packet.
> So a promotion to `uint32_t` instead of `time_t` could be smarter ?

We're all screwed in Jan 2038... I hope to retire by then!

Reply:
2037 will be when the big money to enable retirement will be around :)